AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/StorageEngine+DeleteTransaction.swift

```diff
@@ -43,18 +47,72 @@ extension StorageEngine {
             return .failure(causedBy: error)
         }
 
-        return collapseResults(queryResult: queriedResult, deleteResult: deletedResult)
+        return collapseResults(queryResult: queriedResult, deleteResult: deletedResult, associatedModels: associatedModels)
+    }
+
+    func recurseQueryAssociatedModels(modelSchema: ModelSchema, ids: [Model.Identifier]) -> [Model] {
+        var associatedModels: [Model] = []
+        for (_, modelField) in modelSchema.fields {
+            guard modelField.hasAssociation,
+                modelField.isOneToOne || modelField.isOneToMany,
+                let associatedModelName = modelField.associatedModelName,
+                let associatedField = modelField.associatedField,
+                let associatedModelSchema = ModelRegistry.modelSchema(from: associatedModelName) else {
+                    continue
+            }
+            let queriedModels = queryAssociatedModels(modelName: associatedModelName,
+                                                      associatedField: associatedField,
+                                                      ids: ids)
+            let associatedModelIds = queriedModels.map {$0.id}
+            associatedModels.append(contentsOf: queriedModels)
+            associatedModels.append(contentsOf: recurseQueryAssociatedModels(modelSchema: associatedModelSchema, ids: associatedModelIds))
+        }
+        return associatedModels
     }
 
-    func collapseResults<M: Model>(queryResult: DataStoreResult<[M]>?,
-                                   deleteResult: DataStoreResult<[M]>?) -> DataStoreResult<[M]> {
+    func queryAssociatedModels(modelName: ModelName, associatedField: ModelField, ids: [Model.Identifier]) -> [Model] {
+        guard let modelType = ModelRegistry.modelType(from: modelName) else {
+            log.error("Failed to lookup \(modelName)")
+            return []
+        }
+
+        var queriedModels: [Model] = []
+        let maxNumberOfPredicates = 950
```

Comment:
Any recommendations on how to change this?

Reply:
Some kind of documentation of intent would be nice: is the 950 because we have some upper system bound? We feel it's a reasonable performance throttle? We've tested and this is the upper limit before the code falls over?

Reply:
Ah, got it, I will add a comment.